### ISSUE TITLE
Move to use CLI commands instead of Python script

### ISSUE THREAD

**Issue by nateraw** [nateraw/huggingface-sync-action#6]
might be nice to move to use CLI tools from Hugging Face Hub client library that weren't around when I originally put this repo together. That way we don't have to maintain our own python logic.

Upload Folder: https://huggingface.co/docs/huggingface_hub/v0.26.2/en/guides/cli#upload-an-entire-folder

Notably it takes care of creating the repo, which is 1/2 of what the script here in the repo currently does

**Comment by mrshu** [comment]
@nateraw I presume this will mean installing the CLI via `pip install -U "huggingface_hub[cli]"` and then just calling `huggingface-cli` within `action.yml`, right?

**Comment by nateraw** [comment]
yes I think so. Also if you want to just solve your issue #5 and not do this, that's fine too :). Python fire, which we're using in the script, should handle list of str args so it should be very quick fix.

This one might be a little more involved, so if you don't want to sink time into it don't worry about it :) 

Thanks for your help in this repo I really appreciate it!!

**Comment by mrshu** [comment]
Sorry, only got back to this now.

Looking at the docs, it seems like the `--include` parameter is supported with `huggingface-cli upload` command, so even #5 should be fixed relatively nicely (it might make sense to expose other config variables as well).

I guess the best course of action would not necessarily be running `huggingface-cli` but installing `huggingface_hub` and using the `upload_folder` function (e.g. https://github.com/huggingface/huggingface_hub?tab=readme-ov-file#upload-files).

Would that make sense to you?

Thanks!

**Comment by nateraw** [comment]
Hey there, no worries ❤ 

I think either approach would work. I think I originally thought the CLI would be best because we could get rid of python script altogether, then. But I really am not highly opinionated either way...so whatever is easiest for you.

Obviously haven't had time to look into this myself either, so help in any form would be much appreciated.
